pkg/replica/replica.go

```diff
@@ -232,6 +236,37 @@ func construct(readonly bool, size, sectorSize int64, dir, head string, backingF
 	return r, r.writeVolumeMetaData(true, r.info.Rebuilding)
 }
 
+func (r *Replica) checkVolumeMetaFile() (err error) {
+	filePath := r.diskPath(volumeMetaData)
+	fileInfo, err := os.Stat(filePath)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	if fileInfo != nil && fileInfo.Size() != 0 {
```

Comment:
Don't just use `fileInfo.Size()` to determine if this needs to recover, but check if the content can be decoded to `replica.Info`, so this recovery method can tackle all cases w/ invalid content.

Reply:
Modified, use `r.unmarshalFile()` to check if the content can be decoded to `replica.Info` but only check metadata file is empty or does not exist now